Some minor needed upgrades

jonatjano opened the pull request:
- [x] add all language-todo tags (solve #33 )
- [x] add option to ignore commonly ignored files
- [x] add an option to chose the scope of the search : 
    - [x] add the options in the package configuration
    - [x] use the value of the option value to search the corresponding scope
-  [x] different scopes for the searches (write the functions since I have the idea of how it works) : 
    - [x] search in the active file only *
    - [x] search in every opened files *
    - [x] search in the first project only
    - [x] search in the first project and opened files *
    - [x] search in project linked to opened files * (solve #38 )
    - [x] search in workspace (every project + opened files) *

\* :  should solve #37 (work with file not saved but opened)

jonatjano committed:
added an option to ignore commonly ignored paths

jonatjano committed:
added support for every tag of language-TODO package

jonatjano committed:
corrected a bug where file with x TODO would appear to have x^x TODO since it showed the sames a second time

jonatjano commented:
@reergymerej what's up with travis? the problem is on a line I didn't touch... :upside_down_face: 
Do you have an idea ?

jonatjano committed:
Delete yarn.lock

should have been ignored from the start

jonatjano committed:
completed commonly ignored files array

jonatjano committed:
Merge branch 'master' of https://github.com/jonatjano/todo

reergymerej commented:
I'm not sure.  It may be upset because you deleted yarn.lock.

reergymerej commented:
Error: Cannot find module 'atom'
    at Function.Module._resolveFilename (module.js:476:15)
    at Function.Module._load (module.js:424:25)
    at Module.require (module.js:504:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/travis/build/reergymerej/todo/lib/todo.js:6:1)
    at Module._compile (module.js:577:32)
    at loader (/home/travis/build/reergymerej/todo/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/home/travis/build/reergymerej/todo/node_modules/babel-register/lib/node.js:154:7)

jonatjano commented:
you don't have yarn.lock on your version so I don't think it can be the cause

jonatjano commented:
and for the travis log I can see it too don't worry ;)

reergymerej commented on lib/todo.js:
```diff
@@ -16,7 +24,7 @@ const Todo = {
       description: 'used in conjunction with RegExp Flags to find todo items in your code',
       type: 'string',
-      default: 'TODO\\:.+',
+      default: '(TODO|FIXME|CHANGED|XXX|IDEA|HACK|NOTE|REVIEW|NB|BUG|QUESTION|COMBAK|TEMP)\\:.+',
```
I think this part is really cool and could fit in its own small PR for #33 while we figure out the rest.

jonatjano replied:
since the change is very small maybe you could just modify it using the github editor

jonatjano commented:
Ok I'm going to sleep (It is 01:32 here :sleeping_bed:) I managed to get every files for a project, if I have time tomorrow I'll do the scan and then all that's left will be to find why travis doesn't want what I've done :grinning:

jonatjano committed:
treeview now differentiate package

jonatjano commented:
https://github.com/reergymerej/todo/blob/059c42676df273bbd5391e7b5ff3f0abeb83d695/lib/todo-view.js#L69

do you remember when is this filepath setted ? I can't find it :cry:

jonatjano committed:
TODOs are searched using TextBuffers
allowing for options describing where to search (tmp:search only in the
first project)

jonatjano committed:
test for a better understanding of Travis

jonatjano committed:
previous commit has a sintax error

jonatjano commented:
I still can't understand why Travis fail :/

jonatjano committed:
doesn't compare node by their icons anymore

jonatjano committed:
all wanted behavior are obtained

jonatjano commented:
@reergymerej all that's left is to make the test pass

jonatjano committed:
trying something with babel

jonatjano committed:
trying another thing

jonatjano committed:
trying again

jonatjano committed:
the last one didn't work

jonatjano commented:
@reergymerej I won't be able to search this week-end, if you want you can try all you want.
The package work exactly as expected Travis's so fucking annoying :cry:

jonatjano committed:
deleted that line by mistake :p

reergymerej commented:
Thanks for the effort @jonatjano.  Unfortunately, I'm not really set up to get to this soon.  I'll try to get to it when I get a chance, but I'm not sure how long that will be.  If anyone else has ideas, please let us know.
